### ISSUE TITLE
Unexpected failure during module execution - IPA client test

### ISSUE THREAD

**Issue by foxpluto** [freeipa/ansible-freeipa#181]
Dear developers,

I have a strange error during the execution of ipa client role, here is the output of the tast in very verbose mode: 
```
TASK [ipaclient : Install - IPA client test] *******************************************************************************************************************************************************************
task path: /Users/fox/.ansible/collections/ansible_collections/freeipa/ansible_freeipa/roles/ipaclient/tasks/install.yml:33
<10.0.20.200> ESTABLISH SSH CONNECTION FOR USER: ansible
<10.0.20.200> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="../ssh_user/id_rsa"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="ansible"' -o ConnectTimeout=10 -o ControlPath=/Users/fox/.ansible/cp/e14714b765 10.0.20.200 '/bin/sh -c '"'"'echo ~ansible && sleep 0'"'"''
<10.0.20.200> (0, b'/home/ansible\n', b'')
<10.0.20.200> ESTABLISH SSH CONNECTION FOR USER: ansible
<10.0.20.200> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="../ssh_user/id_rsa"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="ansible"' -o ConnectTimeout=10 -o ControlPath=/Users/fox/.ansible/cp/e14714b765 10.0.20.200 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /home/ansible/.ansible/tmp/ansible-tmp-1577983117.893297-111740363514151 `" && echo ansible-tmp-1577983117.893297-111740363514151="` echo /home/ansible/.ansible/tmp/ansible-tmp-1577983117.893297-111740363514151 `" ) && sleep 0'"'"''
<10.0.20.200> (0, b'ansible-tmp-1577983117.893297-111740363514151=/home/ansible/.ansible/tmp/ansible-tmp-1577983117.893297-111740363514151\n', b'')
The full traceback is:
Traceback (most recent call last):
  File "/usr/local/Cellar/ansible/2.9.2/libexec/lib/python3.7/site-packages/ansible/executor/task_executor.py", line 146, in run
    res = self._execute()
  File "/usr/local/Cellar/ansible/2.9.2/libexec/lib/python3.7/site-packages/ansible/executor/task_executor.py", line 645, in _execute
    result = self._handler.run(task_vars=variables)
  File "/usr/local/Cellar/ansible/2.9.2/libexec/lib/python3.7/site-packages/ansible/plugins/action/normal.py", line 46, in run
    result = merge_hash(result, self._execute_module(task_vars=task_vars, wrap_async=wrap_async))
  File "/usr/local/Cellar/ansible/2.9.2/libexec/lib/python3.7/site-packages/ansible/plugins/action/__init__.py", line 803, in _execute_module
    (module_style, shebang, module_data, module_path) = self._configure_module(module_name=module_name, module_args=module_args, task_vars=task_vars)
  File "/usr/local/Cellar/ansible/2.9.2/libexec/lib/python3.7/site-packages/ansible/plugins/action/__init__.py", line 197, in _configure_module
    **become_kwargs)
  File "/usr/local/Cellar/ansible/2.9.2/libexec/lib/python3.7/site-packages/ansible/executor/module_common.py", line 1258, in modify_module
    environment=environment)
  File "/usr/local/Cellar/ansible/2.9.2/libexec/lib/python3.7/site-packages/ansible/executor/module_common.py", line 1095, in _find_module_utils
    py_module_cache, zf)
  File "/usr/local/Cellar/ansible/2.9.2/libexec/lib/python3.7/site-packages/ansible/executor/module_common.py", line 731, in recursive_finder
    [os.path.join(*py_module_name[:-idx])])
  File "/usr/local/Cellar/ansible/2.9.2/libexec/lib/python3.7/site-packages/ansible/executor/module_common.py", line 650, in __init__
    self.get_source()
  File "/usr/local/Cellar/ansible/2.9.2/libexec/lib/python3.7/site-packages/ansible/executor/module_common.py", line 666, in get_source
    data = pkgutil.get_data(to_native(self._package_name), to_native(self._mod_name + '.py'))
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/pkgutil.py", line 619, in get_data
    spec = importlib.util.find_spec(package)
  File "/usr/local/Cellar/ansible/2.9.2/libexec/lib/python3.7/importlib/util.py", line 114, in find_spec
    raise ValueError('{}.__spec__ is None'.format(name))
ValueError: ansible_collections.freeipa.ansible_freeipa.plugins.module_utils.ansible_ipa_client.__spec__ is None

fatal: [test_server]: FAILED! => {
    "msg": "Unexpected failure during module execution.",
    "stdout": ""
}
```
My installation is: 
```
MacBook-Pro:ansible fox$ ansible --version
ansible 2.9.2
  config file = /Users/fox/.ansible.cfg
  configured module search path = ['/Users/fox/.ansible/collections/ansible_collections/freeipa/ansible_freeipa/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.9.2/libexec/lib/python3.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.6 (default, Dec 30 2019, 19:38:26) [Clang 11.0.0 (clang-1100.0.33.16)]
```

What is wrong here ?

Thanks,
Stefano

**Comment by t-woerner** [comment]
I was now able to reproduce this, but only with virtual env and the Ansible collection. I do not see so far why this happens. It needs further investigation.

**Comment by t-woerner** [comment]
I have created #233 - it was solving an issue with the loading of the ansible_ipa_server, .. in collections for me then I created them in tests with newer ansible versions.

**Comment by seocam** [comment]
@foxpluto can you verify that after #233 this was also fixed for you?

**Comment by t-woerner** [comment]
This is an Ansible issue (ansible/ansible#68361) and will be fixed with Ansible 2.10.
Closing as duplicate of #315
